cdk-lib/src/types.ts

```diff
@@ -0,0 +1,148 @@
+/* eslint-disable @typescript-eslint/naming-convention */
```

Comment:
Documentation for types and props is only partially done here. This should be completed (and have parity with corresponding leo documentation) before publishing.